### PR TITLE
Authenticated Renders

### PR DESCRIPTION
**Pull request by chriskrogh**
## What's changed
Not sure if you noticed in https://github.com/uwaterlooacs/acs/pull/74 but when the user authenticates, we don't want them to see log in / sign up buttons. This PR hides those buttons from the header and from the membership page.

## UI (if UI has changed)
Refer to https://github.com/uwaterlooacs/acs/pull/74

### CODE REVIEW

File: client/src/components/Header/Menu.tsx [hunk 1]
```diff
@@ -31,6 +32,9 @@ interface Props extends WithStyles<typeof styles> {
 }
 
 function Menu({ classes, currentPathname, onLoginClicked }: Props) {
+  const { user } = useContext(UserContext);
+  const authenticated = user._id !== '0';
```
Comment (jtaaa) [inline client/src/components/Header/Menu.tsx:36]: Hmmm maybe we should make `user` possibly undefined or null in the `UserContext` and null/undefined in the initial state. Is there any reason it's populated when the user isn't signed in?

Comment (chriskrogh) [reply client/src/components/Header/Menu.tsx:36]: I'm 100% down for that actually. That would make our lives a lot easier. Will implement

**Comment by jtaaa** [comment]
If you wanna do it in a separate PR I can approve this boy! To avoid conflicts and stuff with the other branches

File: client/src/components/Header/Menu.tsx (hunk 1 above)
Comment (chriskrogh) [reply client/src/components/Header/Menu.tsx]: updated in 48b9d10776c25b5fc24a8411c69180091fed98bf

**Pull request merged**